ocular/src/keyring.rs

```diff
@@ -0,0 +1,740 @@
+        }
+    }
+
+    fn rename_key(
+        &self,
+        current_name: &str,
+        new_name: &str,
+        override_if_exists: bool,
+    ) -> Result<(), KeyStoreError> {
+        // Check if current key exists
+        if !self.key_exists(current_name)? {
+            eprintln!("Key '{}', does not exist.", current_name);
+            return Err(KeyStoreError::DoesNotExist(current_name.to_string()));
+        }
+
+        // Check if new key exists
+        if self.key_exists(new_name)? && !override_if_exists {
+            eprintln!("New key name '{}', already exists.", new_name);
+            return Err(KeyStoreError::Exists(new_name.to_string()));
+        }
+
+        // Proceed with rename
+        let current_name = &KeyName::new(current_name)
+            .unwrap_or_else(|_| panic!("Could not create KeyName for '{}'.", current_name));
+        let new_name = &KeyName::new(new_name)
+            .unwrap_or_else(|_| panic!("Could not create KeyName for '{}'.", new_name));
+
+        let key = self
+            .key_store
+            .as_ref()
+            .expect("Error accessing key store.")
+            .load(current_name)
+            .expect("Could not load key.");
+
+        // Create new key.
+        let _result = self
+            .key_store
+            .as_ref()
+            .expect("Error accessing key store.")
+            .store(new_name, &key);
+
+        // Delete old key.
+        let _result = self.delete_key(current_name);
+
+        Ok(())
+    }
+
+    fn get_key_address(&self, name: &str) -> Result<PublicKeyOutput, KeyStoreError> {
+        // Check if key exists
+        if !self.key_exists(name)? {
+            eprintln!("Key '{}', does not exist.", name);
+            return Err(KeyStoreError::DoesNotExist(name.to_string()));
+        }
+
+        // Prepare key and parse into pem
+        let key_name = &KeyName::new(name)
+            .unwrap_or_else(|_| panic!("Could not create KeyName for '{}'.", name));
+        let key = self
+            .key_store
+            .as_ref()
+            .expect("Error accessing key store.")
+            .load(key_name)
+            .expect("Could not load key");
+        let key: k256::elliptic_curve::SecretKey<k256::Secp256k1> = key
+            .to_pem(LineEnding::default())
+            .expect("Could not convert to pem.")
+            .parse()
+            .expect("Could not parse pem");
+
+        // Convert to deepspace private key for address conversion
+        let key_bytes = key.to_bytes();
+        let key = deep_space::utils::bytes_to_hex_str(&key_bytes);
+        let deep_space_key: deep_space::private_key::PrivateKey =
+            key.parse().expect("Could not parse private key.");
+
+        // Finally get address
+        // TODO: Support other prefixes if necessary
+        let address = deep_space_key
+            .to_address(COSMOS_ADDRESS_PREFIX)
+            .expect("Could not generate address.")
+            .to_bech32(COSMOS_ADDRESS_PREFIX)
```

Comment:
yeah I think we'd probably want this to be configurable if we want to support different chains like that. Based off of my limited knowledge of relayers, i think maybe we want this to be dynamically set from the user. If we're relaying from somm to atom we'd need both somm and atom keys (as I understand it). It might be nice to allow the user to set different key types in the same keystore for convenience. Thoughts?

Reply:
Can continue this on the updated code